bin/latest_backup.sh

```diff
@@ -117,18 +136,180 @@ generate_presigned_urls() {
     local bucket_name=$1
     local files=("$@")
     local urls=()
+
     for file in "${files[@]:1}"; do
-        local url=$(aws s3 presign "s3://${file}" --expires-in 3600)
-        urls+=("$url")
+        # Check if the file exists in the S3 bucket
+        if aws s3 ls "s3://${bucket_name}/${file}" >/dev/null 2>&1; then
+            local url=$(aws s3 presign "s3://${bucket_name}/${file}" --expires-in 3600)
+            urls+=("$url")
+        else
+            echo "Error: File s3://${bucket_name}/${file} does not exist."
+            exit 1
+        fi
     done
+
     echo "${urls[@]}"
 }
 
+# Function to list all ZIP files in the same S3 path as the latest backup
+list_all_zip_files() {
```

Comment:
```sh
sh bin/latest_backup.sh --list-zip-files
```

![Screenshot 2024-07-30 at 9 15 12 AM](https://github.com/user-attachments/assets/28bf0275-2bfa-48e8-b2fa-bcede4d8da34)

Reply:
I dont see that issue? can you try again

Reply:
I am unable to reproduce this error